llvm/lib/Target/AMDGPU/SIWholeQuadMode.cpp

```diff
@@ -1688,25 +1686,32 @@ bool SIWholeQuadMode::runOnMachineFunction(MachineFunction &MF) {
         BuildMI(Entry, EntryMI, DebugLoc(), TII->get(AMDGPU::COPY), LiveMaskReg)
             .addReg(Exec);
     LIS->InsertMachineInstrInMaps(*MI);
+    Changed = true;
   }
 
   LLVM_DEBUG(printInfo());
 
-  lowerLiveMaskQueries();
-  lowerCopyInstrs();
+  Changed |= lowerLiveMaskQueries();
+  Changed |= lowerCopyInstrs();
 
-  // Shader only needs WQM
-  if (GlobalFlags == StateWQM) {
+  if (!(GlobalFlags & ~StateExact)) {
+    // No wave mode execution
+    Changed |= lowerKillInstrs(false);
```

Comment:
Before this patch we never called `lowerKillInstrs(false)`. Was that a bug?

Reply:
No, it wasn't necessary.
We ran the whole lowering for an Exact shader with kills.
In the whole lower `lowerBlock` performs the same function as lowerKillInstrs.